src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/dataobject.cs

```diff
@@ -1677,19 +1677,13 @@ private int SaveObjectToHandle(IntPtr handle, Object data, bool doNotReallocate)
         {
             Stream stream;
             BinaryWriter binaryWriter;
-            BinaryFormatter formatter;
 
             using (stream = new MemoryStream())
             {
                 using (binaryWriter = new BinaryWriter(stream))
                 {
                     binaryWriter.Write(_serializedObjectID);
-
-                    formatter = new BinaryFormatter();
-
-                    #pragma warning disable SYSLIB0011 // BinaryFormatter is obsolete 
-                    formatter.Serialize(stream, data);
-                    #pragma warning restore SYSLIB0011 // BinaryFormatter is obsolete
+                    BinaryFormatWriter.TryWriteFrameworkObject(stream,data);
```

Comment:
The idea here is to try to use the `BinaryFormatWritter` first, then fall back to the `BinaryFormatter` if you cannot for some reason. See our [`DataObject` implementation](https://github.com/dotnet/winforms/blob/dbc7ce32e8bf24466e69c1907af2977f3aee68dc/src/System.Windows.Forms/src/System/Windows/Forms/OLE/DataObject.cs#L653-L671).

You shouldn't remove the `BinaryFormatter` as a fall back unless you know that there are no user scenarios that might need it. While it is depreciated and will likely be disabled by default for WPF/WinForms soon, there will always be a way for users to enable it if they need tol

Reply:
> there will always be a way for users to enable it

That isn’t true. If you use the `BinaryFormatter` anywhere, for anything, you will see member-does-not-exist errors that stop every WPF build come .NET 9. `BinaryFormatter` must be completely removed.

Reply:
@JeremyKuhne Thanks for the review.
I have added ```BinaryFormatter``` as a fall back.

Reply:
> > there will always be a way for users to enable it
> 
> That isn’t true. If you use the `BinaryFormatter` anywhere, for anything, you will see member-does-not-exist errors that stop every WPF build come .NET 9. `BinaryFormatter` must be completely removed.

Can you please source where you're getting your information? I'm positive the APIs are not being removed.

Reply:
I suspect the source is the same place I linked in the dotnet github repo design document

[BinaryFormatter Obsoletion Strategy](https://github.com/dotnet/designs/blob/main/accepted/2020/better-obsoletion/binaryformatter-obsoletion.md#binaryformatter-infrastructure-removed-from-net-net-9)

While it does say it **_may_** get moved to a 3rd party package, it does not guarantee it.

> "All code bases which target net9.0 or above must be fully migrated away from BinaryFormatter at this time. This applies to dotnet org first-party code, partner team code, and third-party user code."

> "The implementation of all such APIs will be to throw PlatformNotSupportedException. The internal BinaryFormatter implementation will no longer exist in .NET in any form and cannot be reenabled through any compat switch."

If this information is false, then perhaps the dotnet team should update the design document to more accurately represent the future plans.